src/pyhf/__init__.py

```diff
@@ -27,24 +28,55 @@ def get_backend():
 
 
 @events.register('change_backend')
-def set_backend(backend, custom_optimizer=None):
+def set_backend(backend, custom_optimizer=None, _session=None):
     """
     Set the backend and the associated optimizer
 
     Example:
         >>> import pyhf
         >>> import tensorflow as tf
-        >>> pyhf.set_backend(pyhf.tensor.tensorflow_backend(session=tf.Session()))
+        >>> sess = tf.compat.v1.Session()
+        >>> pyhf.set_backend("tensorflow", _session=sess)
 
     Args:
-        backend: One of the supported pyhf backends: NumPy, TensorFlow, and PyTorch
+        backend (`str` or `pyhf.tensor` backend): One of the supported pyhf backends: NumPy, TensorFlow, and PyTorch
+        custom_optimizer (`pyhf.optimize` optimizer): Optional custom optimizer defined by the user
+        _session (|tf.compat.v1.Session|_): TensorFlow v1 compatible Session to use when the :code:`"tensorflow"` backend API is used
+
+    .. |tf.compat.v1.Session| replace:: ``tf.compat.v1.Session``
+    .. _tf.compat.v1.Session: https://www.tensorflow.org/api_docs/python/tf/compat/v1/Session
 
     Returns:
         None
     """
     global tensorlib
     global optimizer
 
+    if isinstance(backend, (str, bytes)):
+        if isinstance(backend, bytes):
+            backend = backend.decode("utf-8")
+        # Needed while still using TF v1.0 API
+        if backend == "tensorflow":
+            backend = tensor.tensorflow_backend(session=_session)
+        else:
+            try:
+                backend = getattr(tensor, "{0:s}_backend".format(backend))()
+            except TypeError:
+                raise InvalidBackend(
+                    "\"{0:s}\" is not a supported backend. Select from one of the supported backends: numpy, tensorflow, pytorch".format(
+                        backend
+                    )
+                )
+
+    _name_supported = getattr(tensor, "{0:s}_backend".format(backend.name))
```

Comment:
what's the difference btwn this and line 63 above?

Reply:
Line 63 is:

```python
getattr(tensor, "{0:s}_backend".format(backend))()
```

while this is
```python
getattr(tensor, "{0:s}_backend".format(backend.name))
```

In like 63 `backend` is a string, and in line 71 `backend` is a `pyhf.tensor` object.